### ISSUE TITLE
Developing a download & extract module for SparseBase and, its integration to SparseBase

### ISSUE THREAD

**Issue by SinanEkm** [sparcityeu/sparsebase#135]
Developing a download&extract module for users to save them the trouble of doing this process manually.

-Avoid external libraries such as libarchive due to dependency problems. Instead, use header-only libraries,
-Integrate this module into the SparseBase. Place this module utils/io folder and under the io namespace,
-Create tests

**Comment by AmroAlJundi** [comment]
Maybe for downloaders of specific sites (SuiteSparse for instance,) we can name matrices with specific names depending on their use. For example, the main matrix can always be called "Matrix", and if an additional vector is present, we can call it "Solution". Can make users' experience nicer.

**Comment by SinanEkm** [comment]
> Maybe for downloaders of specific sites (SuiteSparse for instance,) we can name matrices with specific names depending on their use. For example, the main matrix can always be called "Matrix", and if an additional vector is present, we can call it "Solution". Can make users' experience nicer.

I am using unordered_map to keep <matrixName, path> pairs, maybe after I added all the files in the directory I can add the "Matrix" key which has the main matrix path. By the way, as I mentioned in the last meeting, there are some matrices that have multiple files alongside the main matrices, so calling the additional vector "Solution" might not work for those cases, because there might be multiple additional vectors.

**Comment by AmroAlJundi** [comment]
That would make sense. Yeah, you're right. We could scrape the entire database and manually name these different matrices. It's not too crazy of an idea since we're building a class specifically for that data source, but I think we need to know the scale of the database. Do you know how many matrices there are in SuiteSparse?

**Comment by SinanEkm** [comment]
I didn't get the idea. Why do we scrape the entire database? To find the matrices that have multiple files? By the way, there are 2893 matrices in total in the Suitesparse.

**Comment by AmroAlJundi** [comment]
Yes, exactly. We can classify them into categories based on how many matrices there are and give these categories names. At least we can get an idea about how many have more than one matrix.
